src/contracts/protocol/tokenization/RWAAToken.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.10;
+
+import {AccessControl} from 'src/contracts/dependencies/openzeppelin/contracts/AccessControl.sol';
+import {IncentivizedERC20} from 'src/contracts/protocol/tokenization/base/IncentivizedERC20.sol';
+import {SafeCast} from 'src/contracts/dependencies/openzeppelin/contracts/SafeCast.sol';
+import {IERC20} from 'src/contracts/dependencies/openzeppelin/contracts/IERC20.sol';
+import {Errors} from 'src/contracts/protocol/libraries/helpers/Errors.sol';
+import {AToken} from 'src/contracts/protocol/tokenization/AToken.sol';
+import {IRWAAToken} from 'src/contracts/interfaces/IRWAAToken.sol';
+import {IAToken} from 'src/contracts/interfaces/IAToken.sol';
+import {IPool} from 'src/contracts/interfaces/IPool.sol';
+
+abstract contract RWAAToken is AToken, IRWAAToken {
+  using SafeCast for uint256;
+
+  bytes32 public constant ATOKEN_TRANSFER_ROLE = keccak256('ATOKEN_TRANSFER_ROLE');
+
+  /**
+   * @dev Only AToken Transfer Admin can call functions marked by this modifier.
+   */
+  modifier onlyATokenTransferAdmin() {
+    AccessControl aclManager = AccessControl(_addressesProvider.getACLManager());
+    require(
+      aclManager.hasRole(ATOKEN_TRANSFER_ROLE, msg.sender),
+      Errors.CALLER_NOT_ATOKEN_TRANSFER_ADMIN
+    );
+    _;
+  }
+
+  modifier onlyTreasuryRecipient(address recipient) {
+    require(recipient == _treasury, Errors.RECIPIENT_NOT_TREASURY);
+    _;
+  }
+
+  /**
+   * @dev Constructor.
+   * @param pool The address of the Pool contract
+   */
+  constructor(IPool pool) AToken(pool) {
+    // Intentionally left blank
+  }
+
+  /// @inheritdoc AToken
```

Comment:
wondering if we should point to the IRWAAToken instead, given that behavior has changed (all of them revert now)

Reply:
[78c0cfe](https://github.com/aave/aave-v3-horizon/pull/1/commits/78c0cfe181273d953b2a026a5bbbb2dfa152068a)